jscomp/runtime/caml_parser.ml

```diff
@@ -170,40 +170,40 @@ let caml_parse_engine : parse_tables -> parser_env -> (*Parsing.parser_input *)C
     //var shift_recover = 9;
     //var reduce = 10;
     // Parsing.parser_env
-    var env_s_stack = 0; // array
-    var env_v_stack = 1; // array
-    var env_symb_start_stack = 2; // array
-    var env_symb_end_stack = 3; // array
-    var env_stacksize = 4;
-    var env_stackbase = 5;
-    var env_curr_char = 6;
-    var env_lval = 7; // Caml_obj_extern.t
-    var env_symb_start = 8; // position
-    var env_symb_end = 9; // position
-    var env_asp = 10;
-    var env_rule_len = 11;
-    var env_rule_number = 12;
-    var env_sp = 13;
-    var env_state = 14;
-    var env_errflag = 15;
+    var env_s_stack = 's_stack'; // array
+    var env_v_stack = 'v_stack'; // array
+    var env_symb_start_stack = 'symb_start_stack'; // array
```

Comment:
I am thinking can we do some clever things to not encode such invariant in the runtime

Reply:
it seems all fields in Parsing.parse_tables are immutable, so that we can do a field access in the beginning